app/javascript/mastodon/components/status.js

```diff
@@ -119,7 +141,7 @@ export default class Status extends ImmutablePureComponent {
     // and if so, set our isHidden to true to trigger an unrender. The point of
     // this is to save DOM nodes and avoid using up too much memory.
     // See: https://github.com/tootsuite/mastodon/issues/2900
-    this.setState((prevState) => ({ isHidden: !prevState.isIntersecting }));
+    this.setState((prevState, props) => ({ isHidden: !props.visible || !prevState.isIntersecting }));
```

Comment:
I've changed to make "unrendered" state while the status list is hidden, but you can try to disable it (previous behavior) by comment out `!props.visible ||`.

And this implementation probably not be efficient...